### PR TITLE
oneVideo Bid Adapter: adomain & mediaTypes.video Support (SAPR-16874)

### PR DESCRIPTION
**Pull request by adam-browning**
## Type of change
- [ x] Other

## Description of change
Hi @patmmccann 
In alignment with #6807 the oneVideoBidAdapter now listens to bid.mediaTypes.video params first. bid.params.video are used as overrides or in cases where  bid.mediaTypes.video params are missing.
This update includes:
1. First look at bid.mediaTypes.video params
2. bid.params.video - as overrides (backward compatible)
3. Validation granular error messages that will display when pbjs_debug=true.
4. skip validations for e2etest mode.

- test parameters for validating bids
This test will only work if you pass bid.params.site.page = "https:verizonmedia.com"
And make the request using VPN location US west coast.
```
const adUnits = [{
                code: 'video1',
                mediaTypes: {
                    video: {
                        context: 'outstream',
                        playerSize: [300, 250],
                        mimes: ['video/mp4','application/javascript'],
                    }
                },
                bids: [
                    {
                        bidder: 'oneVideo',
                        params: {
                            pubId: 'HBExchange'
                        }
                    }
                ]
            }];
```
## Other information

Hi @DeepthiNeeladri & @kprasadpvv,
Prebid V.5 is supposed to go live on June 9th.
I have created this PR to mitigate removal of the adapter specifically for this version.
Would you be able to review please?
Main changes are in validations and param settings.

Thanks in advance!
Adam

### CODE REVIEW

**Comment by patmmccann** [comment]
Thanks so much! Does this cover #6650 as well?

File: modules/oneVideoBidAdapter.md [hunk 1]
```diff
@@ -298,7 +302,7 @@ var adUnits = [
                     id: 1,
                     page: 'https://verizonmedia.com',
                     referrer: 'https://verizonmedia.com'
-                  },            
+                  },
```
Comment (patmmccann) [inline modules/oneVideoBidAdapter.md:305]: please move openrtb params out of bidder params and into the ad unit in your examples

**Review by patmmccann — CHANGES_REQUESTED** [review]

**Review by DeepthiNeeladri — APPROVED** [review]
Looks good to me!! we are keeping the linearity param the same rest all we are validating

File: modules/oneVideoBidAdapter.md (hunk 1 above)
Comment (adam-browning) [reply modules/oneVideoBidAdapter.md:351]: @patmmccann I have updated the file accordingly

**Comment by adam-browning** [comment]
Hi @patmmccann,
I have added meta.advertiserDomains support that covers  #6650 as well.
Thanks in advance, 
Adam

File: modules/oneVideoBidAdapter.md (hunk 1 above)
Comment (patmmccann) [reply modules/oneVideoBidAdapter.md:351]: Thanks!

**Review by patmmccann — APPROVED** [review]

**Pull request merged**